### ISSUE TITLE
install dependencies?

### ISSUE THREAD

**Issue by BrianInglis** [nexB/license-expression#95]
Python package user not dev. 
Is there any way to update the licence list to current without updating the code? 
I can install 30, but not upgrade to current, because something called hazmat appears to insist on Rust bindings, which I don't care about!
Why do I need to install a Rust compiler to install this module? 
Can pip be run to rollback junk from failed installs/upgrades?

**Comment by pombredanne** [comment]
>  Is there any way to update the licence list to current without updating the code?

Sorry, but not yet, this would be a nice feature though and easy enough,

>  I can install 30, but not upgrade to current, because something called hazmat appears to insist on Rust bindings, which I don't care about!
> Why do I need to install a Rust compiler to install this module?
> Can pip be run to rollback junk from failed installs/upgrades?

I am mystified here by "install a Rust compiler to install this module?", as there is a single dependency at https://github.com/nexB/license-expression/blob/c20b3f605daefc7cd9e4dc7b34e95280f206def3/setup.cfg#L53

Can you paste a log of your installation?

**Comment by BrianInglis** [comment]
Thanks for the quick response. 
Cygwin python39 installed. 
We are trying to get something working for package licence validation in desktop packaging utility, as the sourceware packaging server has an old python 3.6 which can not currently be upgraded (perhaps LTS? Fedora release). 
Initial try:
```
 $ pip3 install license-expression
Requirement already satisfied: license-expression in /usr/local/lib/python3.9/site-packages (30.0.0)
Requirement already satisfied: boolean.py>=4.0 in /usr/local/lib/python3.9/site-packages (from license-expression) (4.0)
```
Simple test program copy/pasted from examples and example programs:
```
$ ./le.py "Junk"
Traceback (most recent call last):
  File "/usr/src/./le.py", line 38, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/usr/src/./le.py", line 21, in main
    parsed = licensing.parse(expression, validate=True)
  File "/usr/local/lib/python3.9/site-packages/license_expression/__init__.py", line 559, in parse
    self.validate_license_keys(expression)
  File "/usr/local/lib/python3.9/site-packages/license_expression/__init__.py", line 466, in validate_license_keys
    raise ExpressionError(msg)
license_expression.ExpressionError: Unknown license key(s): Junk
$ echo "$LICENSE"
MIT AND BSD-2-Clause AND BSD-3-Clause AND BSD-4-Clause-UC AND Linux-man-pages-1-para AND Linux-man-pages-copyleft AND Linux-man-pages-copyleft-var AND Linux-man-pages-copyleft-2-para AND GPL-1.0-or-later AND GPL-2.0-only AND GPL-2.0-or-later AND LGPL-3.0-or-later WITH LGPL-3.0-linking-exception  # man-pages
$ ./le.py "$LICENSE"
Traceback (most recent call last):
  File "/usr/src/./le.py", line 38, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/usr/src/./le.py", line 21, in main
    parsed = licensing.parse(expression, validate=True)
  File "/usr/local/lib/python3.9/site-packages/license_expression/__init__.py", line 559, in parse
    self.validate_license_keys(expression)
  File "/usr/local/lib/python3.9/site-packages/license_expression/__init__.py", line 466, in validate_license_keys
    raise ExpressionError(msg)
license_expression.ExpressionError: Unknown license key(s): Linux-man-pages-1-para, Linux-man-pages-copyleft, Linux-man-pages-copyleft-var, Linux-man-pages-copyleft-2-para
```
Upgrade:
```
$ pip3 install -U license-expression
Requirement already satisfied: license-expression in /usr/local/lib/python3.9/site-packages (30.0.0)
Collecting license-expression
  Downloading license_expression-30.3.0-py3-none-any.whl.metadata (12 kB)
Requirement already satisfied: boolean.py>=4.0 in /usr/local/lib/python3.9/site-packages (from license-expression) (4.0)
Downloading license_expression-30.3.0-py3-none-any.whl (108 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 108.9/108.9 kB 708.5 kB/s eta 0:00:00
Installing collected packages: license-expression
  Attempting uninstall: license-expression
    Found existing installation: license-expression 30.0.0
    Uninstalling license-expression-30.0.0:
      Successfully uninstalled license-expression-30.0.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
scancode-toolkit 32.0.0rc3 requires click!=7.0,>=6.7, which is not installed.
scancode-toolkit 32.0.0rc3 requires colorama>=0.3.9, which is not installed.
scancode-toolkit 32.0.0rc3 requires commoncode>=31.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires container-inspector>=31.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires debian-inspector>=31.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires dparse2>=0.7.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires fasteners, which is not installed.
scancode-toolkit 32.0.0rc3 requires fingerprints>=0.6.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires ftfy>=6.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires gemfileparser2>=0.9.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires intbitset>=3.0.2, which is not installed.
scancode-toolkit 32.0.0rc3 requires jaraco.functools, which is not installed.
scancode-toolkit 32.0.0rc3 requires javaproperties>=0.5, which is not installed.
scancode-toolkit 32.0.0rc3 requires jsonstreams>=0.5.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires packageurl-python>=0.9.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires packvers>=21.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires parameter-expansion-patched>=0.3.1, which is not installed.
scancode-toolkit 32.0.0rc3 requires pdfminer.six>=20200101, which is not installed.
scancode-toolkit 32.0.0rc3 requires pefile>=2020.1.1, which is not installed.
scancode-toolkit 32.0.0rc3 requires pip-requirements-parser>=32.0.1, which is not installed.
scancode-toolkit 32.0.0rc3 requires pkginfo2>=30.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires plugincode>=31.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires publicsuffix2, which is not installed.
scancode-toolkit 32.0.0rc3 requires pyahocorasick>=2.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires pygmars>=0.7.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires pymaven-patch>=0.2.8, which is not installed.
scancode-toolkit 32.0.0rc3 requires saneyaml>=0.6.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires spdx-tools==0.7.0rc0, which is not installed.
scancode-toolkit 32.0.0rc3 requires text-unidecode>=1.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires typecode>=30.0.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires urlpy, which is not installed.
scancode-toolkit 32.0.0rc3 requires xmltodict>=0.11.0, which is not installed.
scancode-toolkit 32.0.0rc3 requires lxml>=4.9.2, but you have lxml 4.7.1 which is incompatible.
Successfully installed license-expression-30.3.0
$ ./le.py "$LICENSE"
None
Traceback (most recent call last):
  File "/usr/src/./le.py", line 39, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/usr/src/./le.py", line 28, in main
    print(parsed.simplify())
AttributeError: 'NoneType' object has no attribute 'simplify'
```
**Assumed** `license-expression` required `scancode-toolkit` (must have been earlier attempt to install *SPDX* tools) and that required those deps (which I attempted to install) which required `Cryptography`>=36 which required `hazmat` Rust bindings! 
Is it just that the [Usage Examples](https://license-expression.readthedocs.io/en/latest/readme_link.html#usage-examples) document needs updated?
